catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitializationTest.java

```diff
@@ -0,0 +1,158 @@
+
+@RunWith(MockitoJUnitRunner.class)
+public class IdentityNodeInitializationTest {
+
+    private static final String TEST_SITE_NAME = "Slate Rock and Gravel Company";
+
+    private static final String TEST_VERSION = "FF 2.0";
+
+    @Mock
+    private FederationAdminServiceImpl federationAdminService;
+
+    private IdentityNodeInitialization identityNodeInitialization;
+
+    private Metacard testMetacard;
+
+    private RegistryTransformer registryTransformer;
+
+    @Mock
+    private Subject subject;
+
+    private XmlParser parser;
+
+    private MetacardMarshaller metacardMarshaller;
+
+    @Before
+    public void setUp() throws Exception {
+        parser = spy(new XmlParser());
+        identityNodeInitialization = spy(new IdentityNodeInitialization());
+        registryTransformer = spy(new RegistryTransformer());
+        metacardMarshaller = spy(new MetacardMarshaller(parser));
+        registryTransformer.setParser(parser);
+        identityNodeInitialization.setRegistryTransformer(registryTransformer);
+        identityNodeInitialization.setMetacardMarshaller(metacardMarshaller);
+        identityNodeInitialization.setFederationAdminService(federationAdminService);
+        System.setProperty(SystemInfo.SITE_NAME, TEST_SITE_NAME);
+        System.setProperty(SystemInfo.VERSION, TEST_VERSION);
+        testMetacard = getTestMetacard();
+    }
+
+    @Test
+    public void initWithNoPreviousEntry() throws Exception {
+        ArgumentCaptor<Metacard> captor = ArgumentCaptor.forClass(Metacard.class);
+        when(federationAdminService.getLocalRegistryIdentityMetacard()).thenReturn(Optional.empty());
+        identityNodeInitialization.init();
+        verify(federationAdminService).addRegistryEntry(captor.capture());
+        Metacard metacard = captor.getValue();
+        assertThat(metacard.getAttribute(Metacard.MODIFIED), notNullValue());
+        assertThat(metacard.getAttribute(Metacard.CREATED), notNullValue());
+        assertThat(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_IDENTITY_NODE)
+                .getValue(), equalTo(true));
+        assertThat(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE)
+                .getValue(), equalTo(true));
+    }
+
+    @Test
+    public void initWithPreviousPrimaryEntry() throws Exception {
+        when(federationAdminService.getLocalRegistryIdentityMetacard()).thenReturn(Optional.of(
+                testMetacard));
+        identityNodeInitialization.init();
+        verify(federationAdminService, never()).addRegistryEntry(any(Metacard.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void initWithGetLocalRegistryNodeException() throws Exception {
+        doThrow(FederationAdminException.class).when(federationAdminService)
+                .getLocalRegistryIdentityMetacard();
+        identityNodeInitialization.init();
+        verify(federationAdminService, never()).addRegistryEntry(any(Metacard.class));
```

Comment:
Don't think you need this line since it will never be run. init will throw the exception.